### PR TITLE
renderBase64 with Quality Parameter

### PR DESCRIPTION
**Pull request by grippy**
See this issue/feature request here:
https://github.com/ariya/phantomjs/issues/12474

### CODE REVIEW

**Comment by Vitallium** [comment]
LGTM

**Comment by ariya** [comment]
Please add some tests.

**Comment by grippy** [comment]
@ariya Just added tests.

File: src/webpage.cpp [hunk 1]
```diff
@@ -979,7 +979,7 @@ bool WebPage::render(const QString &fileName, const QVariantMap &option)
     return retval;
 }
 
-QString WebPage::renderBase64(const QByteArray &format)
+QString WebPage::renderBase64(const QByteArray &format, const int &quality)
```
Comment (ariya) [inline src/webpage.cpp]: It's `int`, no need to pass by reference.

File: src/webpage.h [hunk 2]
```diff
@@ -253,9 +253,10 @@ public slots:
      *
      * @brief renderBase64
      * @param format String containing one of the supported types
+     * @param quality containing a value from 1 to 100. Default is -1.
      * @return Rendering base-64 encoded of the page if the given format is supported, otherwise an empty string
      */
-    QString renderBase64(const QByteArray &format = "png");
+    QString renderBase64(const QByteArray &format = "png", const int &quality = -1);
```
Comment (ariya) [inline src/webpage.h]: ditto.

**Comment by ariya** [comment]
Please squash the commits together.

**Comment by grippy** [comment]
@ariya a few changed based on your feedback:

- Commits squashed
- quality no longer passed by reference

**Comment by grippy** [comment]
@ariya Is this ready to be merged?

**Comment by flesler** [comment]
:+1:

**Comment by zackw** [comment]
We need you to update the tests to the new test harness.  Also, what happens if someone calls renderBase64 without passing a quality parameter?  That needs to keep working.

**Comment by grippy** [comment]
@zackw If you don't pass a quality, it'll default to -1. I'm done with this PR. Should've been merged along time ago IMHO. No reason to have it sit this long for such a small change.

**Comment by jampy** [comment]
Any chance this will be merged soon?

**Pull request merged**